lib/routes/user/add_funds/add_funds_page.dart

```diff
@@ -120,121 +168,195 @@ class AddFundsState extends State<AddFundsPage> {
-              border: Border.all(
-                  color: Colors.white, style: BorderStyle.solid, width: 1.0),
-              borderRadius: BorderRadius.circular(14.0)),
-          child: Row(
-            mainAxisSize: MainAxisSize.min,
-            mainAxisAlignment: MainAxisAlignment.spaceBetween,
-            children: <Widget>[
-              Padding(
-                padding: EdgeInsets.only(left: 8.0, right: 4.0),
-                child: Image(
-                  image: AssetImage("src/icon/vendors/fastbitcoins_logo.png"),
-                  height: 24.0,
-                  fit: BoxFit.scaleDown,
-                  color: theme.fastbitcoins.iconFgColor,
-                ),
-              ),
-              Expanded(
-                child: Padding(
-                  padding: EdgeInsets.only(right: 8.0),
-                  child: AutoSizeText(
-                    'REDEEM FASTBITCOINS VOUCHER',
-                    style: theme.fastbitcoinsTextStyle,
-                    maxLines: 1,
-                    minFontSize: MinFontSize(context, fontSize: theme.fastbitcoinsTextStyle.fontSize).minFontSize,
-                    stepGranularity: 0.1,
-                  ),
-                ),
-              )
-            ],
+  Widget _buildDepositToBTCAddress() {
+    return GestureDetector(
+      behavior: HitTestBehavior.translucent,
+      child: Container(
+        height: 72,
+        width: MediaQuery.of(context).size.width,
+        child: Row(mainAxisAlignment: MainAxisAlignment.start, children: <Widget>[
+          Padding(
+            padding: const EdgeInsets.only(left: 16, right: 16),
+            child: Image(
+              image: AssetImage("src/icon/bitcoin.png"),
+              height: 24.0,
+              width: 24.0,
+              fit: BoxFit.scaleDown,
+              color: Colors.white,
+            ),
+          ),
+          Expanded(
+            child: Text(
+              'DEPOSIT TO BTC ADDRESS',
+              style: theme.addFundsItemsStyle,
+            ),
           ),
-        ));
+          Padding(padding: const EdgeInsets.all(8.0), child: Icon(Icons.keyboard_arrow_right, color: Colors.white, size: 24.0)),
+        ]),
+      ),
+      onTap: () => Navigator.push(
+        context,
+        FadeInRoute(
+          builder: (_) => DepositToBTCAddressPage(widget._user, widget._accountBloc),
+        ),
+      ),
+    );
   }
 
-  Widget _buildBottomBar(AddFundResponse response, AccountModel account,
-      {hasError = false}) {
-    if (hasError || response?.errorMessage?.isNotEmpty == true) {
-      return SingleButtonBottomBar(
-          text: hasError ? "RETRY" : "CLOSE",
-          onPressed: () {
-            if (hasError) {
-              _addFundsBloc.addFundRequestSink.add(null);
-            } else {
-              Navigator.of(context).pop();
-            }
-          });
+  Widget _buildMoonpayButton(AddFundResponse response) {
+    String baseUrl = "https://buy-staging.moonpay.io";
```

Comment:
The moonpay url should be generated in the bloc. Also the `apiKey` should be fetched from a config file.

Reply:
Fixed
[1](https://github.com/breez/breezmobile/pull/162/commits/e2cbce256ee9aeba8b56b024e9eb6c650dbbb913)
[2](https://github.com/breez/breezmobile/pull/162/commits/2c95697004ad429b7857223898348aa9ab1ec3c0)